### DISCUSSION TITLE
Python script that reverses a string input from the user and handles empty input?

### DISCUSSION THREAD

**Discussion by avagoji** [community/community#134675] [Other Feature Feedback, Questions, & Ideas] [answered]
### Select Topic Area

Question

### GitHub Feature Area

Actions

### Body

How do you create a Python script that reverses a string input from the user and handles empty input?

**Comment by suprscott** [comment, marked as answer]
**Code:**
```
def reverse_string(s):
    return s[::-1]

def main():
    user_input = input("Enter a string to reverse: ")
    
    if not user_input:  # Check if the input is empty
        print("Error: No input provided. Please enter a valid string.")
    else:
        reversed_str = reverse_string(user_input)
        print(f"The reversed string is: '{reversed_str}'")

if __name__ == "__main__":
    main()
```

### **Explanation**
**1. Function Definition:**
- The reverse_string function takes a string s and returns it reversed using Python's slicing feature.

**2. Main Function:**
- Prompts the user to enter a string.
- Checks if the input is empty with if not user_input:. If it is, it prints an error message.
- If the input is valid, it calls the reverse_string function and prints the reversed string.
